File: implementacao/src/models/entities/Animal.ts
class Animal {
    constructor(
        name: string,
        especie: string,
        endereco: string,
        descricao: string,
        filhote: boolean,
        status: string,
        id: number,
        img: string

    ) {
        this.name = name;
        this.especie = especie;
        this.endereco = endereco;
        this.descricao =  descricao;
        this.filhote = filhote;
        this.status = status; 
        this.id = id;
        this.img = img;
    }

    name: string;
    especie: string;
    endereco: string;
    descricao: string;
    filhote: boolean;
    status: string;
    id: number;
    img: string;

    mudaStatus(status: string) {
        this.status = status;
    }
}

export default Animal;

File: app/app/src/main/java/com/app/myapplication/model/entitie/Animal.kt
package com.app.myapplication.model.entitie

data class Animal(
    var nome:String,
    var status: String,
    var idade: Int,
    var raca: String,
    var gato : Boolean,
    var cachorro: Boolean,
    var filhote: Boolean,
    var endereco: String,
    var descricao: String,
    var cliente: Cliente
){
    constructor() : this("", "", -1, "", true, false, false, "", "", Cliente())
    fun mudarStatus(status: String){
        this.status = status
    }

    fun notificarCliente(status: String) {
        // enviar email
        print(cliente.email.endereco)
    }
}